lib/ansible/modules/cloud/azure/azure_rm_trafficmanagerprofile.py

```diff
@@ -0,0 +1,451 @@
+        default: enabled
+        choices:
+            - enabled
+            - disabled
+    routing_method:
+        description:
+            - The traffic routing method of the Traffic Manager profile.
+        default: performance
+        choices:
+            - performance
+            - priority
+            - weighted
+            - geographic
+    dns_config:
+        description:
+            - The DNS settings of the Traffic Manager profile.
+        suboptions:
+            relative_name:
+                description:
+                    - The relative DNS name provided by this Traffic Manager profile.
+                    - If not provided, name of the Traffic Manager will be used
+            ttl:
+                description:
+                    - The DNS Time-To-Live (TTL), in seconds.
+                default: 60
+    monitor_config:
+        description:
+            - The endpoint monitoring settings of the Traffic Manager profile.
+        suboptions:
+            protocol:
+                description:
+                    - The protocol (HTTP, HTTPS or TCP) used to probe for endpoint health.
+                choices:
+                    - HTTP
+                    - HTTPS
+                    - TCP
+            port:
+                description:
+                    - The TCP port used to probe for endpoint health.
+            path:
+                description:
+                    - The path relative to the endpoint domain name used to probe for endpoint health.
+            interval:
+                description:
+                    - The monitor interval for endpoints in this profile in seconds.
+            timeout:
+                description:
+                    - The monitor timeout for endpoints in this profile in seconds.
+            tolerated_failures:
+                description:
+                    - The number of consecutive failed health check before declaring an endpoint in this profile Degraded after the next failed health check.
+        default:
+            protocol: HTTP
+            port: 80
+            path: /
+
+extends_documentation_fragment:
+    - azure
+    - azure_tags
+
+author:
+    - "Hai Cao <t-haicao@microsoft.com>"
+    - "Yunge Zhu <yungez@microsoft.com>"
+
+'''
+
+EXAMPLES = '''
+    - name: Create a Traffic Manager Profile
+      azure_rm_trafficmanager:
+        name: tmtest
+        resource_group: tmt
+        location: global
+        profile_status: enabled
+        routing_method: priority
+        dns_config:
+          relative_name: tmtest
+          ttl: 60
+        monitor_config:
+          protocol: HTTPS
+          port: 80
+          path: '/'
+        tags:
+          Environment: Test
+
+    - name: Delete a Traffic Manager Profile
+      azure_rm_trafficmanager:
+        state: absent
+        name: tmtest
+        resource_group: tmt
+'''
+RETURN = '''
+id:
+    description: The ID of the traffic manager profile
+    returned: always
```

Comment:
The code shows that this is only returned when `state: present`. I would change the returned key to be `traffic manager profile exists`. Update the code to always return this and the endpoints list when `state: present` and when `state: absent` and you actually removed an endpoint.

Reply:
updated